src/odemis/gui/cont/cryo_project.py

```
@@ -0,0 +1,154 @@
+        if feature.path:
+            feature_item['path'] = feature.path
+        feature_list.append(feature_item)
+
+    overview_list = serialize_images(overviews)
+    return {"version": PROJECT_VERSION, "features": feature_list, "overviews": overview_list}
+
+def add_image(images, filename, indices=None):
+    # Our naming schemes should not allow to add a duplicate filename, so that is not handled here
+    images.append({IMG_FILENAME: filename, **({IMG_IN_FILE_IDS: set(indices)} if indices else {})})
+
+def remove_image(images, filename: os.PathLike, indices: Iterable = None):
```

Ruff (0.15.10)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)